src/Sema.zig

```diff
@@ -15988,46 +15988,45 @@ fn analyzeArithmetic(
     };
 
     try sema.requireRuntimeBlock(block, src, runtime_src);
-    if (block.wantSafety() and want_safety and scalar_tag == .Int) {
+
+    if (block.wantSafety() and want_safety and scalar_tag == .Int) blk: {
+        const op_ov_tag: Air.Inst.Tag = switch (air_tag) {
+            .add => .add_with_overflow,
+            .sub => .sub_with_overflow,
+            .mul => .mul_with_overflow,
+            else => break :blk,
```

Comment:
This case could include `assert(air_tag == air_tag_safe);` to avoid it getting out of sync with the beginning of the function, or that condition could be included in the if allowing this to be `else => unreachable,` which is less of a code smell.

Reply:
To optimise `analyzeArithmetic`, remove `air_tag` and `air_tag_safe` from the main switch.
Compute `air_tag` like:
```zig
const air_tag: Air.Inst.Tag = switch (zir_tag) {
    .add, .add_unsafe => if (block.float_mode == .Optimized) .add_optimized else .add,
    .addwrap => .add_wrap,
    .add_sat => .add_sat,
    .sub => if (block.float_mode == .Optimized) .sub_optimized else .sub,
    .subwrap => .sub_wrap,
    .sub_sat => .sub_sat,
    .mul => if (block.float_mode == .Optimized) .mul_optimized else .mul,
    .mulwrap => .mul_wrap,
    .mul_sat => .mul_sat,
    else => unreachable,
};
```
Compute entry to the runtime safety block like:
```zig
if (block.wantSafety() and switch (air_tag) {
    .add, .sub, .mul => want_safety,
    else => false,
}) {
```
Compute `air_tag_safe` and resolve `panic` like:
```zig
if (mod.backendSupportsFeature(.safety_checked_instructions)) {
    _ = try sema.preparePanicId(block, .integer_overflow);
    return block.addBinOp(switch (air_tag) {
        .add => .add_safe,
        .sub => .sub_safe,
        else => .mul_safe,
    }, casted_lhs, casted_rhs);
}
```
It is up to you. The only purpose of this PR is to fix the miscompilation.

Reply:
I'm not talking about an optimization, I'm just talking about avoiding a load-bearing else prong, instead of making even more logic depend on it.  Your new comment contains a significantly worse else prong, which suggests that I'm not explaining properly.

For example, if you start with:
```
const value: enum { a, b, c } = f();
switch (value) {
    .a => a(),
    .b => b(),
    else => c(),
}
```
Now, adding an enum tag causes the code to be silently wrong.  However, if you use an exhaustive switch instead, or explicitly handle the cases that the code currently supports and have an `else => unreachable,`, then adding an enum tag can at least points you to the problem instead of being actively unhelpful.

Anyway, the simplest improvement would be:
```suggestion
            .add_unsafe, .addwrap, .add_sat, .subwrap, .sub_sat, .mulwrap, .mul_sat => break :blk,
            else => unreachable,
```

Reply:
I do not understand why you want me to rewrite this switch for you now. It has been in master for well over a year and was edited without issue this September. I am not adding anything unrelated to the bug-fix that will require a CI re-run.